cfe_internal/update/update_policy.cf

```diff
@@ -300,13 +298,106 @@ bundle agent cfe_internal_update_policy_cpv
+      "cfe_internal_update_cmdb_data_consumption";
+}
+
+bundle agent cfe_internal_update_cmdb_data_distribution
+# @brief Ensure data is ready for agents to download
+{
+  vars:
+    policy_server.enterprise_edition::
+
+      # The API response for host specific data from cmdb tells us the timestamp of the last data change
+      # We store this timestamp and use it for the next request.
+      "_cmdb_next_request_state_file"
+        string => "$(sys.statedir)/cmdb_next_request_from.dat";
+
+      # If we have the timestamp from a previous response we use it, else we start from 0
+      "_cmdb_previous_next_request_from"
+        string => readfile( $(_cmdb_next_request_state_file), inf ),
+        if => regline( "^\d+$", $(_cmdb_next_request_state_file) );
+
+      "_cmdb_previous_next_request_from"
+        string => "0",
+        unless => regline( "^\d+$", $(_cmdb_next_request_state_file) );
+
+      # We need a script to call that should return the API response
+      "_get_cmdb_data_bin" string => "$(sys.bindir)/get_cmdb.php";
+      "_get_cmdb_data_cmd" string => "/var/cfengine/httpd/php/bin/php $(_get_cmdb_data_bin) $(_cmdb_previous_next_request_from)";
+
+      # We call the script and we pass it the timestamp from the prior call
+      "_get_cmdb_data_response"
+        string => execresult( $(_get_cmdb_data_cmd), useshell ), # This would be execresult()
+        if => fileexists( $(_get_cmdb_data_bin) );
+
+      "_get_cmdb_data_response_d"
+        data => mergedata('$(_get_cmdb_data_response)');
+
+      # So that we can write a JSON file for each host we get the indicies of data in the response
+      "_i" slist => getindices( "_get_cmdb_data_response_d[data]");
+
+      # We need to store the timestamp from the most recent change so that we can use that as a starting point for future requests.
+      "_next_request_from"
+        string => "$(_get_cmdb_data_response_d[meta][cmdb_epoch])";
+
   files:
-      "$(sys.workdir)/data/." create => "true";
+      # "$(_get_cmdb_data_cmd)" perms => m( 700 );
+
+      # If the next request state file doesn't exist, we seed one with 0, the lowest epoch value possible. because we populate variables from this file content.
+      "$(_cmdb_next_request_state_file)"
+        create => "true",
+        handle => "cmdb_data_change_next_seed",
+        template_data => parsejson('{}'),
+        template_method => "inline_mustache",
+        edit_template_string => "0",
+        unless => fileexists("$(_cmdb_next_request_state_file)" );
+
+      # Write out the data for each host that had a data change
+      "$(sys.workdir)/cmdb/$(_i)/host_specific.json"
+        create => "true", # CFE-2329, ENT-4792
+        template_data => mergedata("_get_cmdb_data_response_d[data][$(_i)]" ),
+        template_method => "inline_mustache",
+        edit_template_string => string_mustache( "{{$-top-}}", mergedata("_get_cmdb_data_response_d[data][$(_i)]" ) ),
```

Comment:
Is `mergedata()` needed here?

Reply:
I /think/ it is needed to "pick" out the substructure.

Reply:
Oh, turns out I don't *need* it.

    bundle agent __main__
    {
       vars:
          "d" data => '{
        "data": {
            "KEY1": {
                "value":"value"
            },
            "KEY2": {
                "value":"2"
            }
        }
    }';
    
      reports:
        "string_mustache @() to subkey$(const.n) $(with)" with => string_mustache("{{%-top-}}", "@(d[data][KEY1])" );
        "string_mustache direct variablename to subkey$(const.n)$(with)" with => string_mustache("{{%-top-}}", "d[data][KEY1]" );
        "mergedata subkkey$(const.n)$(with)" with => string_mustache("{{%-top-}}", mergedata("d[data][KEY1]") );
    
    }

    R: string_mustache @() to subkey
     {
      "value": "value"
    }
    R: string_mustache direct variablename to subkey
    {
      "value": "value"
    }
    R: mergedata subkkey
    {
      "value": "value"
    }

    {
        "data": {
            "KEY1": {
                "value":"value"
            },
            "KEY2": {
                "value":"2"
            }
        }
    }

Reply:
> Oh, turns out I don't _need_ it.

Yeah, I think I even new where in the code this magic happens to work. :) (`VariableResolve2()`)